pilot/pkg/security/model/authentication.go

```diff
@@ -205,8 +209,38 @@ func ConstructValidationContext(rootCAFilePath string, subjectAltNames []string)
 	return ret
 }
 
+// ConstructSANWithTrustDomain constructs MatchSubjectAltNames to be used in DefaultValidationContext by merging
+// Exact SAN matcher user has supplied at Gateway and Prefix SAN matcher for trust domain validation
+func ConstructSANWithTrustDomain(in []string, trustDomainAliases []string) []*matcher.StringMatcher {
+	//  initially contains only usersupplied SAN
+	userSupplied := util.StringToExactMatch(in)
+	// Early exit if Trust Domain not to be verified
+	if features.SkipValidateTrustDomain.Get() {
```

Comment:
Let's check the flag early in `pilot/pkg/networking/plugin/authn/authentication.go` before calling NewPolicyApplier(). If the flag is true then just pass an empty trust domain list.

Reply:
not sure how this would help us, it actually makes code readability a lot worse. Is there a good use of moving this flag check early? And how does moving it early eliminate the input parameter `in` for `constructSANWithTrustDomain`

Reply:
1. It's the same rational for removing the `isGateway` parameter, let the caller to decide what to do in a higher level and let the function behavior to only base on its input parameters. This makes it easier to infer what is being done in this function since it just simplify converts the trustDomainAliases to a list of StringMatcher.

2. You don't need `in` because this function can just generate a list of StringMatcher from the `trustDomainAliases`, and you append the list to the `matchSAN` you had in ApplyToCommonTLSContext(). In other words, instead of repeating the `util.StringToExactMatch()` in two places, it can just be done once in ApplyToCommonTLSContext.